velox/exec/TableScan.cpp

```diff
@@ -92,6 +92,7 @@ RowVectorPtr TableScan::getOutput() {
 
       if (!split.hasConnectorSplit()) {
         noMoreSplits_ = true;
+        pendingDynamicFilters_.clear();
```

Comment:
I think it's still more clear that we clear the pending filters after passing them to data source

Reply:
`pendingDynamicFilters_` is used in `preload` to intitalize different data source object for preloading splits, I think it should be clear after no more splits. Thanks.

Reply:
Emm I see what you want to do.  But merging filters on every split could cause performance regression. I think we need to do it a little differently.  Maybe when we do the preload, we should pass in the scan spec from the current `dataSource_` to use in preloading data source as well.

Reply:
Hello @Yuhta ,
I tried this proposal, the `scanSpec_` can't be used by multi threads same time, even only the `filter_` field in `ScanSpec`. 
For your concern about merging filters on every split, I add two APIs in `DataSource` to get and set the filter for the specific `outputChannel`, then we only need to set the merged filters to the preloading data sources. This should avoid this problem.
Please help to chekc again, thanks.